File: python/spec/aoc2025/day_01_spec.py
#!/usr/bin/env python3
#
from mamba import description, context, it
from expects import expect, equal

import spec.support
from aoc2025 import SecretEntrance

with description('SecretEntrance'):
  with context('Sample'):
    with before.all:
      self.subject = SecretEntrance(
          spec.support.file_to_list(__file__.replace('spec.py', 'sample.txt')))

    with description('Part 1'):

      with it('sample data agrees'):
        expect(self.subject.zeroes()).to(equal(3))

    with description('Part 2'):

      with it('sample data agrees'):
        expect(self.subject.clicks()).to(equal(6))


  with context('Actual'):
    with before.all:
      self.subject = SecretEntrance(
          spec.support.file_to_list(__file__.replace('spec.py', 'actual.txt')))

    with description('Part 1'):

      with it('actual answer above'):
        print(self.subject.zeroes())

      with it('results survive'):
        expect(self.subject.zeroes()).to(equal(1139))

    with description('Part 2'):

      with it('actual answer above'):
        print(self.subject.clicks())

      with it('results survive'):
        expect(self.subject.clicks()).to(equal(6684))

File: ruby/spec/aoc2025/day_01_spec.rb
module Aoc2025
  RSpec.describe SecretEntrance do
    context "Sample" do
      before(:all) do
        @subject = SecretEntrance.new Support.file_to_list __FILE__.gsub('spec.rb', 'sample.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "dunsels" do
          subject.zeroes
        end

        it "sample data agrees" do
          expect(subject.zeroes).to eq 3
        end
      end

      describe "Part 2" do
        it "sample data agrees" do
          expect(subject.clicks).to eq 6
        end
      end
    end

    context "Actual" do
      before(:all) do
        @subject = SecretEntrance.new Support.file_to_list __FILE__.gsub('spec.rb', 'actual.txt')
      end
      subject { @subject }

      describe "Part 1" do
        it "actual answer above" do
          pp subject.zeroes
        end

        it "results survive" do
          expect(subject.zeroes).to eq 1139
        end
      end

      describe "Part 2" do
        it "actual answer above" do
          pp subject.clicks
        end

        it "results survive" do
          expect(subject.clicks).to eq 6684
        end
      end
    end
  end
end

